### DISCUSSION TITLE
sccache-dist generates a lot of traffic?

### DISCUSSION THREAD

**Discussion by to266** [mozilla/sccache#2201] [Q&A]
I have attempted to use `sccache-dist` form a "normal" home network to fast machines, and it seemed super slow when used with rust. Upon some inspection I realized that my bottleneck was actually the internet bandwidth - the local client was using all of my upload (40 Mbps). 

My suspicion is that the toolchains are somehow not being reused and uploaded anew with every crate that needs compiling? As there can easily be hundreds of crates for a substantial project (and the more crates, the more sense there is to use `sccache` in the first place), this rather explodes in network traffic. 

Is my observation somewhat correct, or did I just somehow hit a corner case?

I might have the resources to attempt fixing this, if there's some guidance for that, too

**Comment by trxcllnt** [comment]
The sccache client doesn't use a request queue for sending toolchains. If you kick off many parallel jobs at once, they'll all try to send the same toolchain to the build server. This is obviously not ideal, and something I'm looking into fixing.
